helm/cluster-openstack/templates/_helpers.tpl

```diff
@@ -42,6 +41,50 @@ room for such suffix.
 ubuntu-2004-kube-v{{ .Values.kubernetesVersion }}
 {{- end -}}
 
+{{- define "hash" -}}
+{{- $inputs := (dict
+  "cloudConfig" .Values.cloudConfig
+  "cloudName" .Values.cloudName
+  "baseDomain" .Values.baseDomain
+  "imageName" (include "imageName" .)
+  "externalNetworkID" .Values.externalNetworkID
+  "rootVolume" .Values.rootVolume
+  "nodeClasses" .Values.nodeClasses
+  "controlPlane" (dict
+    "machineFlavor" .Values.controlPlane.machineFlavor
+    "diskSize" .Values.controlPlane.diskSize)
+  "oidc" .Values.oidc) }}
+{{- mustToJson $inputs | toString | quote | sha1sum | trunc 8 }}
+{{- end -}}
```

Comment:
Here's the most important part. Generate a global hash of all input values that affect any template CRs. This hash can then be appended to all template names. Changing any input will lead to a new hash giving template rotation managed by helm.